python/valis/db/db.py

```diff
@@ -19,9 +19,13 @@
 
 
 def reset_db_state():
-    """ Sub-depdency for get_db that resets the context connection state """
-    pdb._state._state.set(db_state_default.copy())
-    pdb._state.reset()
+    """ Sub-dependency for get_db that resets the context connection state """
+
+    from valis.main import settings
+
+    if settings.db_reset:
+        pdb._state._state.set(db_state_default.copy())
+        pdb._state.reset()
 
 
```

Comment:
in development mode, wrapping this back into a Dependency makes the first query slow since that's when it's first establishing the connection, but then all subsequent queries are fast.  I think this is ok.